packages/tools/replay-tool/src/replayMessages.ts

```diff
@@ -69,6 +70,34 @@ function expandTreeForReadability(tree: ITree): ITree {
     return newTree;
 }
 
+function compareSnapshotStrings(content1: string, content2: string) {
+    const snapshot1: IFileSnapshot = JSON.parse(content1);
+    const snapshot2: IFileSnapshot = JSON.parse(content2);
+
+    deepCompareSnapshots(snapshot1.tree, snapshot2.tree);
+
+    const keys1 = Object.keys(snapshot1.commits);
+    const keys2 = Object.keys(snapshot1.commits);
+    strict.strictEqual(keys1.length, keys2.length, "Snapshot commit lengths do not match");
+    strict.strictEqual(keys1.join(""), keys2.join(""), "Snapshot commit keys do not match");
+
+    for (const commit of Object.keys(snapshot1.commits)) {
+        deepCompareSnapshots(snapshot1.commits[commit], snapshot2.commits[commit]);
+    }
+}
+
+function sameContent(content1: ContainerContent, content2: ContainerContent): boolean {
+    assert(content1.op === content2.op);
+
+    try {
```

Comment:
I assume we have try/catch here and asserts deeper in the code only to ease debugging, right?
I wonder if there are better ways to achieve the same without using exceptions. IN my experience, saving files to disk and using windiff (i.e. visually looking at differences) was fastest and most rebelieving tool to debug issues, so I never had a need to do the same through debugging - it's harder and less ambient information is available to make it efficient. 

If you go with exceptions, can you please change "compare" to "validate" lingvo in names of functions?

Reply:
Writing out files though assumes that they are sorted (fully) prior to comparison / writing to disk. I.e. it's not sort-as-you-go technique, but rather sort whole file, and then do comparison.